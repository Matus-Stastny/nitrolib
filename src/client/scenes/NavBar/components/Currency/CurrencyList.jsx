// @flow strict
import * as React from "react";
import styled from "styled-components";
import { graphql, createFragmentContainer } from "react-relay";

import Flex from "client/primitives/Flex";
import Tooltip from "client/components/Tooltip";
import separateList from "client/services/utils/separateList";
import type { Currencies, Currency } from "client/records/Currency";
import { brandDefault } from "client/records/Brand";
import { BORDER_RADIUS } from "client/consts/layout";
import mq from "client/services/utils/mediaQuery";
import CurrencyItem from "./CurrencyItem";
import Code from "./Code";
import Name from "./Name";
import Sign from "./Sign";
import type { CurrencyList_list } from "./__generated__/CurrencyList_list.graphql";

const COLUMNS = 4;

const Column = styled.div`
  width: 100%;

  ${mq.gtTablet`
    width: ${100 / COLUMNS}%;
  `};
`;

const Item = styled.div`
  ${mq.gtTablet`
    padding-right: 25px;
  `};
`;

const ItemText = styled.div`
  line-height: 30px;
  padding-left: 5px;
  border-radius: ${BORDER_RADIUS}px;
  cursor: pointer;
  background: ${props => (props.active ? props.theme.colors["primary-600"] : "transparent")};

  &:hover {
    background: ${props => props.theme.colors[props.active ? "primary-600" : "grey-100"]};
  }

  ${Code} {
    font-weight: bold;
    color: ${props => props.theme.colors[props.active ? "white" : "grey-900"]};
  }

  ${Sign} {
    margin-left: 10px;
    color: ${props => props.theme.colors[props.active ? "white" : "grey-700"]};
  }

  ${Name} {
    margin-left: 10px;
    color: ${props => props.theme.colors[props.active ? "white" : "inherit"]};
  }

  ${mq.gtTablet`
    ${Name} {
      display: none;
    }
  `};
`;

ItemText.defaultProps = {
  theme: brandDefault.theme,
};

const Tip = styled.span`
  white-space: nowrap;
`;

type Props = {|
  // eslint-disable-next-line react/no-unused-prop-types
  active: Currency,
  available: Currencies,
  list: CurrencyList_list,
  // eslint-disable-next-line react/no-unused-prop-types
  onSetCurrency: (code: string) => void,
|};

const CurrencyList = (props: Props) => (
  <Flex wrap="wrap">
    {separateList(
      COLUMNS,
      // FIXME filter directly in GraphQL
      (props.list.edges &&
        props.list.edges
          .map(edge => edge && edge.node)
          .filter(Boolean)
          .filter(node => props.available[String(node.code)])) ||
        [],
    ).map((items, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Column key={i}>
        {items.map(item => (
          <Item key={item.code}>
            <Tooltip position={i > 1 ? "left" : "right"} tip={<Tip>{item.name}</Tip>}>
              <ItemText
                active={item.code === props.active.id}
                onClick={() => props.onSetCurrency(item.code || "")}
              >
                {/* $FlowIssue */}
                <CurrencyItem item={item} />
              </ItemText>
            </Tooltip>
          </Item>
        ))}
      </Column>
    ))}
  </Flex>
);

export const CurrencyListUnwrapped = CurrencyList;

export default createFragmentContainer(
  CurrencyList,
  graphql`
    fragment CurrencyList_list on CurrencyDetailConnection {
      edges {
        node {
          code
          name
          ...CurrencyItem_item
        }
      }
    }
  `,
);
