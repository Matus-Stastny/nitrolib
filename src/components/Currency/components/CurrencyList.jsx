// @flow strict
import * as React from "react";
import styled, { css } from "styled-components";

import type { Currency } from "../../../records/Currency";
import { themeDefault } from "../../../records/Theme";
import type { ThemeProps } from "../../../records/Theme";
import separateList from "../../../services/utils/separateList";
import Flex from "../../../primitives/Flex";
import mq from "../../../styles/mq";
import * as rtl from "../../../styles/rtl";
import Tooltip from "../../Tooltip";
import CurrencyItem from "./CurrencyItem";
import Code from "./Code";
import Name from "./Name";
import Sign from "./Sign";

type ActiveProps = {|
  ...ThemeProps,
  active: boolean,
|};

const COLUMNS = 4;

const Column = styled.div`
  width: 100%;

  ${mq.gtTablet(css`
    width: ${100 / COLUMNS}%;
  `)};
`;

const Item = styled.div`
  ${mq.gtTablet(css`
    padding-${rtl.right}: 25px;
  `)};
`;

Item.defaultProps = {
  theme: themeDefault,
};

const ItemText = styled.div`
  display: flex;
  line-height: 30px;
  padding-${rtl.left}: 5px;
  border-radius: ${({ theme }: ThemeProps) => theme.orbit.borderRadiusNormal};
  cursor: pointer;
  background: ${({ theme, active }: ActiveProps) =>
    active ? theme.orbit.paletteProductNormal : "transparent"};

  &:hover {
    background: ${({ theme, active }: ActiveProps) =>
      theme.orbit[active ? "paletteProductNormal" : "paletteCloudNormal"]};
  }

  ${Code} {
    font-weight: bold;
    color: ${({ theme, active }: ActiveProps) =>
      theme.orbit[active ? "paletteWhite" : "paletteInkNormal"]};
  }

  ${Sign} {
    margin-${rtl.left}: 10px;
    color: ${({ theme, active }: ActiveProps) =>
      theme.orbit[active ? "paletteWhite" : "paletteInkNormal"]};
  }

  ${Name} {
    margin-${rtl.left}: 10px;
    color: ${({ theme, active }: ActiveProps) =>
      theme.orbit[active ? "paletteWhite" : "paletteInkNormal"]};
  }

  ${mq.gtTablet(css`
    ${Name} {
      display: none;
    }
  `)};
`;

ItemText.defaultProps = {
  theme: themeDefault,
};

const Tip = styled.span`
  white-space: nowrap;
`;

type Props = {|
  active: Currency,
  list: Currency[],
  onSetCurrency: (code: string) => void,
|};

const CurrencyList = ({ active, list, onSetCurrency }: Props) => (
  <Flex wrap="wrap">
    {separateList(COLUMNS, list).map((items, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Column key={i}>
        {items.map(item => (
          <Item key={item.id} data-test={`CurrencySwitcher-Item-${item.id}`}>
            <Tooltip position={i > 1 ? "left" : "right"} tip={<Tip>{item.name}</Tip>}>
              <ItemText active={item.id === active.id} onClick={() => onSetCurrency(item.id)}>
                <CurrencyItem item={item} />
              </ItemText>
            </Tooltip>
          </Item>
        ))}
      </Column>
    ))}
  </Flex>
);

export default CurrencyList;
