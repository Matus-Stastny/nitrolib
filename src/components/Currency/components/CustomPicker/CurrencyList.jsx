// @flow strict
import * as React from "react";
import styled, { css } from "styled-components";

import Tooltip from "../../../Tooltip";
import type { Currency } from "records/Currency";
import { themeDefault } from "records/Theme";
import separateList from "services/utils/separateList";
import Flex from "primitives/Flex";
import mq from "styles/mediaQuery";
import CurrencyItem from "./CurrencyItem";
import Code from "./Code";
import Name from "./Name";
import Sign from "./Sign";

const COLUMNS = 4;

const Column = styled.div`
  width: 100%;

  ${mq.gtTablet(css`
    width: ${100 / COLUMNS}%;
  `)};
`;

const Item = styled.div`
  ${mq.gtTablet(css`
    padding-right: 25px;
  `)};
`;

const ItemText = styled.div`
  line-height: 30px;
  padding-left: 5px;
  border-radius: 3px;
  cursor: pointer;
  background: ${({ theme, active }) => (active ? theme.orbit.paletteProductNormal : "transparent")};

  &:hover {
    background: ${({ theme, active }) => theme.orbit[active ? "primary-600" : "grey-100"]};
  }

  ${Code} {
    font-weight: bold;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "grey-900"]};
  }

  ${Sign} {
    margin-left: 10px;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "grey-700"]};
  }

  ${Name} {
    margin-left: 10px;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "inherit"]};
  }

  ${mq.gtTablet(css`
    ${Name} {
      display: none;
    }
  `)};
`;

ItemText.defaultProps = {
  theme: themeDefault,
};

const Tip = styled.span`
  white-space: nowrap;
`;

type Props = {|
  active: Currency,
  list: Currency[],
  onSetCurrency: (code: string) => void,
|};

const CurrencyList = ({ active, list, onSetCurrency }: Props) => (
  <Flex wrap="wrap">
    {separateList(COLUMNS, list).map((items, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Column key={i}>
        {items.map(item => (
          <Item key={item.id}>
            <Tooltip position={i > 1 ? "left" : "right"} tip={<Tip>{item.name}</Tip>}>
              <ItemText active={item.id === active.id} onClick={() => onSetCurrency(item.id)}>
                <CurrencyItem item={item} />
              </ItemText>
            </Tooltip>
          </Item>
        ))}
      </Column>
    ))}
  </Flex>
);

export default CurrencyList;
