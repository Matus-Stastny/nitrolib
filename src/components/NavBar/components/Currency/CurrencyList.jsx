// @flow strict
import * as React from "react";
import styled, { css } from "styled-components";
import { graphql, createFragmentContainer } from "react-relay";

import Tooltip from "components/Tooltip";
import type { Currencies, Currency } from "records/Currency";
import { themeDefault } from "records/Theme";
import separateList from "services/utils/separateList";
import Flex from "primitives/Flex";
import mq from "styles/mediaQuery";
import CurrencyItem from "./CurrencyItem";
import Code from "./Code";
import Name from "./Name";
import Sign from "./Sign";
import type { CurrencyList_list } from "./__generated__/CurrencyList_list.graphql";

const COLUMNS = 4;

const Column = styled.div`
  width: 100%;

  ${mq.gtTablet(css`
    width: ${100 / COLUMNS}%;
  `)};
`;

const Item = styled.div`
  ${mq.gtTablet(css`
    padding-right: 25px;
  `)};
`;

const ItemText = styled.div`
  line-height: 30px;
  padding-left: 5px;
  border-radius: 3px;
  cursor: pointer;
  background: ${({ theme, active }) => (active ? theme.orbit.paletteProductNormal : "transparent")};

  &:hover {
    background: ${({ theme, active }) => theme.orbit[active ? "primary-600" : "grey-100"]};
  }

  ${Code} {
    font-weight: bold;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "grey-900"]};
  }

  ${Sign} {
    margin-left: 10px;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "grey-700"]};
  }

  ${Name} {
    margin-left: 10px;
    color: ${({ theme, active }) => theme.orbit[active ? "white" : "inherit"]};
  }

  ${mq.gtTablet(css`
    ${Name} {
      display: none;
    }
  `)};
`;

ItemText.defaultProps = {
  theme: themeDefault,
};

const Tip = styled.span`
  white-space: nowrap;
`;

type Props = {|
  // eslint-disable-next-line react/no-unused-prop-types
  active: Currency,
  available: Currencies,
  list: CurrencyList_list,
  // eslint-disable-next-line react/no-unused-prop-types
  onSetCurrency: (code: string) => void,
|};

const CurrencyList = ({ active, available, list, onSetCurrency }: Props) => (
  <Flex wrap="wrap">
    {separateList(
      COLUMNS,
      // FIXME filter directly in GraphQL
      (list.edges &&
        list.edges
          .map(edge => edge && edge.node)
          .filter(Boolean)
          .filter(node => available[String(node.code)])) ||
        [],
    ).map((items, i) => (
      // eslint-disable-next-line react/no-array-index-key
      <Column key={i}>
        {items.map(item => (
          <Item key={item.code}>
            <Tooltip position={i > 1 ? "left" : "right"} tip={<Tip>{item.name}</Tip>}>
              <ItemText
                active={item.code === active.id}
                onClick={() => onSetCurrency(item.code || "")}
              >
                {/* $FlowIssue */}
                <CurrencyItem item={item} />
              </ItemText>
            </Tooltip>
          </Item>
        ))}
      </Column>
    ))}
  </Flex>
);

export const CurrencyListUnwrapped = CurrencyList;

export default createFragmentContainer(
  CurrencyList,
  graphql`
    fragment CurrencyList_list on CurrencyDetailConnection {
      edges {
        node {
          code
          name
          ...CurrencyItem_item
        }
      }
    }
  `,
);
