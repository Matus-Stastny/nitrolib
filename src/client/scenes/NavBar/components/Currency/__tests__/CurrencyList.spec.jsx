// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import CurrencyList from "../CurrencyList";

const currencies = {
  dkk: {
    id: "dkk",
    name: "Danish Krone",
    format: "__price__ kr",
    uncertainFormat: true,
    round: "0",
    enabledOnAffilId: "",
    fallback: "",
    rate: 0.13434,
  },
  gbp: {
    id: "gbp",
    name: "British Pound Sterling",
    format: "£__price__",
    uncertainFormat: false,
    round: "2",
    enabledOnAffilId: "",
    fallback: "",
    rate: 1.14355,
  },
  eur: {
    id: "eur",
    name: "Euro",
    format: "__price__ €",
    uncertainFormat: false,
    round: "2",
    enabledOnAffilId: "",
    fallback: "",
    rate: 1,
  },
  czk: {
    id: "czk",
    name: "Czech Republic Koruna",
    format: "__price__ Kč",
    uncertainFormat: false,
    round: "0",
    enabledOnAffilId: "",
    fallback: "",
    rate: 0.0387945,
  },
  hkd: {
    id: "hkd",
    name: "Hong Kong dollar",
    format: "HK$__price__",
    uncertainFormat: false,
    round: "0",
    enabledOnAffilId: "AFFIL",
    fallback: "gbp",
    rate: 0.109077,
  },
  huf: {
    id: "huf",
    name: "Hungarian Forint",
    format: "__price__ Ft",
    uncertainFormat: false,
    round: "0",
    enabledOnAffilId: "AFFIL",
    fallback: "czk",
    rate: 0.00312772,
  },
};

const active = {
  id: "eur",
  name: "Euro",
  format: "__price__ €",
  uncertainFormat: false,
  round: "2",
  enabledOnAffilId: "",
  fallback: "",
  rate: 1,
};

describe("#Currency/CurrencyList", () => {
  test("render", () => {
    const wrapper = shallow(
      <CurrencyList currencies={currencies} active={active} setCurrency={jest.fn()} />,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("CurrencyList__ItemText[active=true]").shallow()).toMatchSnapshot();
    expect(
      wrapper
        .find("CurrencyList__ItemText[active=false]")
        .first()
        .shallow(),
    ).toMatchSnapshot();
  });

  test("handle click", () => {
    const setCurrency = jest.fn();

    const wrapper = shallow(
      <CurrencyList currencies={currencies} active={active} setCurrency={setCurrency} />,
    );

    wrapper
      .find("CurrencyList__Item")
      .first()
      .find("CurrencyList__ItemText")
      .simulate("click");

    expect(setCurrency).toBeCalledWith("dkk");
  });
});
