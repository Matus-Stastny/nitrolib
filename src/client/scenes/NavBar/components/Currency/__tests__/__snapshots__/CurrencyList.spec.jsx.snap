// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Currency/CurrencyList render 1`] = `
<Flex
  direction="row"
  wrap="wrap"
  x="flex-start"
  y="flex-start"
>
  <CurrencyList__Column
    key="0"
  >
    <CurrencyList__Item
      key="eur"
    >
      <Tooltip
        inline={false}
        mobile={false}
        position="right"
        tip={
          <CurrencyList__Tip>
            Euro
          </CurrencyList__Tip>
        }
      >
        <CurrencyList__ItemText
          active={true}
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "accent-100": "#F8CA32",
                "accent-200": "#F8CA32",
                "accent-300": "#F8CA32",
                "accent-400": "#F8CA32",
                "accent-500": "#ffcc00",
                "accent-600": "#F8CA32",
                "accent-700": "#F8CA32",
                "accent-800": "#F8CA32",
                "accent-900": "#F8CA32",
                "brand-info": "",
                "danger-100": " #ffc7c7",
                "danger-200": " #ffa8a8",
                "danger-300": " #ff7e7e",
                "danger-400": " #ff5858",
                "danger-500": "#ff2424",
                "danger-600": " #e51f1f",
                "danger-700": "#d21c1c",
                "danger-800": " #b91919",
                "danger-900": "#9e1616",
                "facebook-500": "#4465ad",
                "facebook-600": "#3b5998",
                "facebook-800": "#344d85",
                "google-500": "#ff5542",
                "google-600": "#f34a38",
                "google-800": "#e84735",
                "grey-100": "#edeff2",
                "grey-200": "#d5dee7",
                "grey-400": "#adb9c5",
                "grey-500": "#94a2b0",
                "grey-600": "#7c8b99",
                "grey-700": "#57626c",
                "grey-800": "#2e353b",
                "grey-900": "#171b1e",
                "insurance-plus": "#F5922A",
                "mail-button-500": "#0097A9",
                "mail-footer-500": "#0097A9",
                "mail-header-500": "#0097A9",
                "neutral-100": "#e9eef2",
                "neutral-200": "#d5dee7",
                "neutral-300": "#c6d0dc",
                "neutral-400": "#adb9c5",
                "neutral-500": "#94a2b0",
                "neutral-600": "#7c8b99",
                "neutral-700": "#57626c",
                "neutral-800": "#2e353b",
                "neutral-900": "#171b1e",
                "primary-100": "#b2f3eb",
                "primary-200": "#77ebdd",
                "primary-300": "#4de3d0",
                "primary-400": "#00d7bd",
                "primary-500": "#00ccb4",
                "primary-600": "#01bba5",
                "primary-700": "#00ad98",
                "primary-800": "#009b88",
                "primary-900": "#008676",
                "success-100": "#d3fac5",
                "success-200": "#b2f79c",
                "success-300": "#8fec70",
                "success-400": "#6ee445",
                "success-500": "#52cf26",
                "success-600": "#37c207",
                "success-700": "#2eb300",
                "success-800": "#29a000",
                "success-900": "#248e00",
                "text-primary": "#ffffff",
                "text-secondary": "#2f2f2f",
                "trip-end": "#d50c6a",
                "trip-start": "#0097a9",
                "warning-100": "#fff1cf",
                "warning-200": "#ffe7b2",
                "warning-300": "#ffdb8d",
                "warning-400": "#ffce67",
                "warning-500": "#ffc345",
                "warning-600": "#f0b539",
                "warning-700": "#dda734",
                "warning-800": "#c9972e",
                "warning-900": "#ac8127",
                "white": "#ffffff",
              },
            }
          }
        >
          <Relay(CurrencyItem)
            item={
              Object {
                "code": "eur",
                "name": "Euro",
              }
            }
          />
        </CurrencyList__ItemText>
      </Tooltip>
    </CurrencyList__Item>
  </CurrencyList__Column>
  <CurrencyList__Column
    key="1"
  >
    <CurrencyList__Item
      key="huf"
    >
      <Tooltip
        inline={false}
        mobile={false}
        position="right"
        tip={
          <CurrencyList__Tip>
            Hungarian Forint
          </CurrencyList__Tip>
        }
      >
        <CurrencyList__ItemText
          active={false}
          onClick={[Function]}
          theme={
            Object {
              "colors": Object {
                "accent-100": "#F8CA32",
                "accent-200": "#F8CA32",
                "accent-300": "#F8CA32",
                "accent-400": "#F8CA32",
                "accent-500": "#ffcc00",
                "accent-600": "#F8CA32",
                "accent-700": "#F8CA32",
                "accent-800": "#F8CA32",
                "accent-900": "#F8CA32",
                "brand-info": "",
                "danger-100": " #ffc7c7",
                "danger-200": " #ffa8a8",
                "danger-300": " #ff7e7e",
                "danger-400": " #ff5858",
                "danger-500": "#ff2424",
                "danger-600": " #e51f1f",
                "danger-700": "#d21c1c",
                "danger-800": " #b91919",
                "danger-900": "#9e1616",
                "facebook-500": "#4465ad",
                "facebook-600": "#3b5998",
                "facebook-800": "#344d85",
                "google-500": "#ff5542",
                "google-600": "#f34a38",
                "google-800": "#e84735",
                "grey-100": "#edeff2",
                "grey-200": "#d5dee7",
                "grey-400": "#adb9c5",
                "grey-500": "#94a2b0",
                "grey-600": "#7c8b99",
                "grey-700": "#57626c",
                "grey-800": "#2e353b",
                "grey-900": "#171b1e",
                "insurance-plus": "#F5922A",
                "mail-button-500": "#0097A9",
                "mail-footer-500": "#0097A9",
                "mail-header-500": "#0097A9",
                "neutral-100": "#e9eef2",
                "neutral-200": "#d5dee7",
                "neutral-300": "#c6d0dc",
                "neutral-400": "#adb9c5",
                "neutral-500": "#94a2b0",
                "neutral-600": "#7c8b99",
                "neutral-700": "#57626c",
                "neutral-800": "#2e353b",
                "neutral-900": "#171b1e",
                "primary-100": "#b2f3eb",
                "primary-200": "#77ebdd",
                "primary-300": "#4de3d0",
                "primary-400": "#00d7bd",
                "primary-500": "#00ccb4",
                "primary-600": "#01bba5",
                "primary-700": "#00ad98",
                "primary-800": "#009b88",
                "primary-900": "#008676",
                "success-100": "#d3fac5",
                "success-200": "#b2f79c",
                "success-300": "#8fec70",
                "success-400": "#6ee445",
                "success-500": "#52cf26",
                "success-600": "#37c207",
                "success-700": "#2eb300",
                "success-800": "#29a000",
                "success-900": "#248e00",
                "text-primary": "#ffffff",
                "text-secondary": "#2f2f2f",
                "trip-end": "#d50c6a",
                "trip-start": "#0097a9",
                "warning-100": "#fff1cf",
                "warning-200": "#ffe7b2",
                "warning-300": "#ffdb8d",
                "warning-400": "#ffce67",
                "warning-500": "#ffc345",
                "warning-600": "#f0b539",
                "warning-700": "#dda734",
                "warning-800": "#c9972e",
                "warning-900": "#ac8127",
                "white": "#ffffff",
              },
            }
          }
        >
          <Relay(CurrencyItem)
            item={
              Object {
                "code": "huf",
                "name": "Hungarian Forint",
              }
            }
          />
        </CurrencyList__ItemText>
      </Tooltip>
    </CurrencyList__Item>
  </CurrencyList__Column>
</Flex>
`;

exports[`#Currency/CurrencyList render 2`] = `
.c0 {
  line-height: 30px;
  padding-left: 5px;
  border-radius: 3px;
  cursor: pointer;
  background: #01bba5;
}

.c0:hover {
  background: #01bba5;
}

.c0 .Code-s9apmq5-0 {
  font-weight: bold;
  color: #ffffff;
}

.c0 .Sign-d8mumz-0 {
  margin-left: 10px;
  color: #ffffff;
}

.c0 .Name-s18tzvjn-0 {
  margin-left: 10px;
  color: #ffffff;
}

@media (min-width:769px) {
  .c0 .Name-s18tzvjn-0 {
    display: none;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <Relay(CurrencyItem)
    item={
      Object {
        "code": "eur",
        "name": "Euro",
      }
    }
  />
</div>
`;

exports[`#Currency/CurrencyList render 3`] = `
.c0 {
  line-height: 30px;
  padding-left: 5px;
  border-radius: 3px;
  cursor: pointer;
  background: transparent;
}

.c0:hover {
  background: #edeff2;
}

.c0 .Code-s9apmq5-0 {
  font-weight: bold;
  color: #171b1e;
}

.c0 .Sign-d8mumz-0 {
  margin-left: 10px;
  color: #57626c;
}

.c0 .Name-s18tzvjn-0 {
  margin-left: 10px;
}

@media (min-width:769px) {
  .c0 .Name-s18tzvjn-0 {
    display: none;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <Relay(CurrencyItem)
    item={
      Object {
        "code": "huf",
        "name": "Hungarian Forint",
      }
    }
  />
</div>
`;
